### DISCUSSION TITLE
Can I Sanitize inputs in Validators (V5)?

### DISCUSSION THREAD

**Discussion by CodeArtisT75** [adonisjs/core#3314] [Help]
Hi.

Can I sanitize user inputs in Validators before validating them in the schema?

something like:

```javascript
export default class ProfileValidator {
  public santize = schema.santize({
    username: 'to_lower_case'
  });

  public schema = schema.create({
    username: schema.string({ trim: true })
  });
}
```

or another way to sanitize them inside the validator (outside of the controller).

thanks.

**Comment by thetutlage** [comment]
You can just create a custom rule and mutate the value inside it. Here's the guide to creating custom rules. https://docs.adonisjs.com/guides/validator/custom-rules

This is how roughly the rule will look like.

```ts
import { string } from '@ioc:Adonis/Core/Helpers'
import { validator } from '@ioc:Adonis/Core/Validator'

validator.rule('toLowerCase', (value, _, options) => {
  if (typeof value !== 'string') {
    return
  }

  options.mutate(value.toLowerCase())
})
```

**Reply by TemaSM** [reply]
I think this example would be a good one to add somewhere to the documentation page you posted a link to
